TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/command_runner.py

```diff
@@ -29,11 +33,15 @@ def run(self) -> CommandResult:
         return self._run_as_synchronous_process()
 
     def sanitized_command(self) -> List[str]:
-        if (not self.password_field) or (self.password_field not in self.command_args):
+        if not self.sensitive_fields:
             return self.command
         # Check whether the value of the password field is in the command
-        password_index = self.command.index(self.command_args[self.password_field])
-        return self.command[:password_index] + ["*" * 8] + self.command[password_index + 1:]
+        display_command = self.command.copy()
+        for field in self.sensitive_fields:
+            if field in display_command:
+                password_index = display_command.index(self.command_args[field])
```

Comment:
This is a bit unintuitive, we're taking the fields as inputs, but looking for the value and replacing it (instead of reconstructing the command_args dict and reconstructing the display command).

There's a slight bug in this logic in that we're sensitizing the first value that we find. e.g.

Given args: `--username admin --password admin`
Our Display_Command will be `--username ******** --password admin`

Reply:
Oh, great point. I'll add that as a test and rework it.